File: agriTech/src/main/java/com/example/agriTech/model/Sensor.java
package com.example.agriTech.model;

import java.time.LocalDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;

@Entity
@Table(name = "sensor_logs")
@Data
public class Sensor {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(name = "sensor_id")
    private String sensorId;
    //DHT11
    @Column(name = "temperature") 
    private Double temperature; // nhiet do
    @Column(name = "humidity")
    private Double humidity;
    //do am dat
    @Column(name = "soil_moisture")
    private Integer soilMoisture; // do am dat
    @Column(name = "status")
    private String status; // "KHO" hoặc "UOT"
    @Column(name = "recorded_at")
    private LocalDateTime recordedAt = LocalDateTime.now(); // du lieu gui lien tuc

   

    @ManyToOne
    @JoinColumn(name = "device_id")
    private Device device;
    
}

File: agriTech(DA3K)/app/src/main/java/com/example/agritechda3k/model/Sensor.kt
package com.example.agritechda3k.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "sensor_data")
data class Sensor(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val deviceId: Long,
    val sensorId: String?,
    val soilMoisture: Int, // Chốt thằng này để vẽ biểu đồ
    val humidity: Double?,

    val temperature: Double?,

    val status: String?,

    val recordedAt: String // Lưu dạng String "yyyy-MM-dd HH:mm:ss" để dễ mapping
)
